### PR TITLE
Prepare for account-pocalypse

### PR DESCRIPTION
**Pull request by smetl**
task:
https://www.odoo.com/web#id=1917430&action=333&active_id=967&model=project.task&view_type=form&menu_id=4720

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by qdp-odoo** [comment]
@robodoo r+ rebase-ff

File: addons/account/models/account_payment_term.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    @api.one
+    def _check_lines(self):
+        payment_term_lines = self.line_ids.sorted()
+        if payment_term_lines and payment_term_lines[-1].value != 'balance':
+            raise ValidationError(_('The last line of a Payment Term should have the Balance type.'))
+        lines = self.line_ids.filtered(lambda r: r.value == 'balance')
+        if len(lines) > 1:
+            raise ValidationError(_('A Payment Term should have only one line of type Balance.'))
+
+    @api.one
```
Comment (Yenthe666) [inline addons/account/models/account_payment_term.py]: Where we not going to stop with api.one? ;-)

Comment (smetl) [reply addons/account/models/account_payment_term.py]: @Yenthe666 The goal is to move stuff, not to refactor the code.

Comment (Yenthe666) [reply addons/account/models/account_payment_term.py]: Still a good moment to touch it in master, not? You're creating a diff already so why not do two things at once 😄 Just my five cents!

Comment (smetl) [reply addons/account/models/account_payment_term.py]: @Yenthe666 Ok ok... you won ;)

Comment (Yenthe666) [reply addons/account/models/account_payment_term.py]: @smetl Awesome, I own you a beer on the next OEXP 😉

**Comment by qdp-odoo** [comment]
@robodoo r+

**Comment by qdp-odoo** [comment]
@robodoo rebase-ff

**Comment by smetl** [comment]
@robodoo retry

**Comment by KangOl** [comment]
@robodoo retry

**Pull request closed without merging**